gitops/snapshot.go

```diff
@@ -775,3 +775,23 @@ func ResetSnapshotStatusConditions(adapterClient client.Client, ctx context.Cont
 
 	return nil
 }
+
+func CopySnapshotLabelsAndAnnotation(application *applicationapiv1alpha1.Application, snapshot *applicationapiv1alpha1.Snapshot, componentName string, source metav1.ObjectMeta, destination metav1.ObjectMeta, prefix string) (*applicationapiv1alpha1.Snapshot, error) {
+
+	if snapshot.Labels == nil {
+		snapshot.Labels = make(map[string]string)
```

Comment:
Wouldn't just this work instead of make() ?
```suggestion
		snapshot.Labels = map[string]string{}
```

Reply:
I think so, updated.